app/scripts/metamask-controller.js

```diff
@@ -6111,23 +6296,16 @@ export default class MetamaskController extends EventEmitter {
       outStream,
       (err) => {
         // handle any middleware cleanup
-        engine._middleware.forEach((mid) => {
-          if (mid.destroy && typeof mid.destroy === 'function') {
-            mid.destroy();
-          }
-        });
+        engine.destroy();
         connectionId && this.removeConnection(origin, connectionId);
-        if (err) {
+        // For context and todos related to the error message match, see https://github.com/MetaMask/metamask-extension/issues/26337
+        if (err && !err.message?.match('Premature close')) {
           log.error(err);
         }
       },
     );
-
-    // Used to show wallet liveliness to the provider
-    if (subjectType !== SubjectType.Internal) {
```

Comment:
?

Reply:
This was likely left in by mistake when we introduced this separate multichain API rpc pipeline - `setupProviderConnectionCaip`. 
a) there is no `chainChanged` event on this API
b) there is injected provider to share liveness with.


@jiexi correct me if I'm wrong on any of this